pkg/apis/flowcontrol/fuzzer/fuzzer.go

```diff
@@ -21,27 +21,33 @@ import (
 
 	runtimeserializer "k8s.io/apimachinery/pkg/runtime/serializer"
 	"k8s.io/kubernetes/pkg/apis/flowcontrol"
+	"k8s.io/utils/ptr"
 )
 
 // Funcs returns the fuzzer functions for the flowcontrol api group.
 var Funcs = func(codecs runtimeserializer.CodecFactory) []interface{} {
 	return []interface{}{
 		func(obj *flowcontrol.LimitedPriorityLevelConfiguration, c fuzz.Continue) {
 			c.FuzzNoCustom(obj) // fuzz self without calling this function again
+
+			// tbd: setting a zero value here will cause the roundtrip
+			// test (from internal to v1beta2, v1beta1 will fail)
+			// question: should we add the roundtrip annotation to
+			// v1beta2, and v1beta1 as well?
+			if obj.NominalConcurrencyShares == nil || *obj.NominalConcurrencyShares == 0 {
+				obj.NominalConcurrencyShares = ptr.To(int32(1))
+			}
```

Comment:
this change makes the roundtrip test less effective since we are no longer testing roundtrip compatibility with a value of zero

Shouldn't we add the defaulting logic for v1beta3 to v1beta2 and v1beta1 as well? When a v1beta1/v1beta2 client  reads a v1beta3/v1 object with a zero value:
- a v1beta1 client requests the object with a zero value
- the object with the roundtrip annotation gets read from the storage and serialized into v1beta3
- it gets converted to internal (we drop the annotation)
- the internal object gets converted to v1beta1 (the annotation is missing) 
- the v1beta1 client gets the object, and it sends an update request to the server (the annotation is missing in the update request)

Reply:
oh, `v1beta2` will be removed in 1.29

Reply:
@wojtek-t : I do not think we need to make a distinction now, and @tkashem , I think the test is appropriate for now.  According to the plan that @liggitt outlined, a pointer-to-zero value will not appear in the NominalConcurrencyShares field until a later release happens. Yes, once we start working on that release then it would be good if we could fuzz for round-trip tests that _should_ allow a pointer-to-zero here as well as fuzz for round-trip tests with v1beta{1,2}. Or do we have a chance to filter in the consumer of the fuzzing?

Reply:
for fuzzing, this is fine

we should have specific non-fuzzed tests exercising every edge of the {unset,zero,non-zero} x {zero-feature disabled, enabled} x {v1beta3,v1} x {create,update} matrix

Reply:
I added these tests here: https://github.com/kubernetes/kubernetes/pull/121089/commits/9fd2ab419ad771790d3cb80ea7b8e6828d9ce305#diff-3e63044c9e6e0118fe9400449384f34e3125f54482d926af14b5e46a2555f6a0